### PR TITLE
Better document the stats module.

### PR DESCRIPTION
**Pull request by ltratt**

### CODE REVIEW

File: ykrt/src/jitstats.rs [hunk 1]
```diff
@@ -17,7 +14,14 @@ use std::{
 };
 use strum::{Display, EnumCount, EnumIter, IntoEnumIterator};
 
+/// Record yk statistics if enabled. In non-testing mode this is only enabled if the end user
+/// defines the global variable `YKD_JITSTATS`. In testing mode this is always enabled, with output
```
Comment (vext01) [inline ykrt/src/jitstats.rs:18]: s/global/environment/ ?

File: ykrt/src/jitstats.rs [hunk 2]
```diff
@@ -17,7 +14,14 @@ use std::{
 };
 use strum::{Display, EnumCount, EnumIter, IntoEnumIterator};
 
+/// Record yk statistics if enabled. In non-testing mode this is only enabled if the end user
```
Comment (vext01) [inline ykrt/src/jitstats.rs:17]: comma after `mode`?

File: ykrt/src/jitstats.rs (hunk 1 above)
Comment (ltratt) [reply ykrt/src/jitstats.rs]: Fixed in e83ddd3.

File: ykrt/src/jitstats.rs (hunk 2 above)
Comment (ltratt) [reply ykrt/src/jitstats.rs:17]: I think it's fine with or without?

File: ykrt/src/jitstats.rs [hunk 3]
```diff
@@ -17,7 +14,14 @@ use std::{
...
+/// defines the global variable `YKD_JITSTATS`. In testing mode this is always enabled, with output
+/// being sent to `stderr`.
```
Comment (vext01) [inline ykrt/src/jitstats.rs]: What's the justification for "always enabled by default in testing?". I'd expect it off by default like all the other things.

Comment (ltratt) [reply ykrt/src/jitstats.rs]: Hmm, that's a good question. You're right, it probably shouldn't be always on.

If it's OK, I'll fix that in the next PR.

File: ykrt/src/jitstats.rs (hunk 2 above)
Comment (vext01) [reply ykrt/src/jitstats.rs:17]: Up to you, but I was jarred on first read.

Comment (vext01) [reply ykrt/src/jitstats.rs]: thanks

File: ykrt/src/jitstats.rs (hunk 3 above)
Comment (vext01) [reply ykrt/src/jitstats.rs]: Please do :)

**Comment by vext01** [comment]
Please squash

**Comment by ltratt** [comment]
Squashed.

File: ykrt/src/jitstats.rs (hunk 3 above)
Comment (vext01) [reply ykrt/src/jitstats.rs]: In the next PR, can it also be a `0/1`-valued variable for consistency?

Comment (ltratt) [reply ykrt/src/jitstats.rs]: That wouldn't make sense -- you have to give a filename (or `-` for stderr) as an argument.

Comment (vext01) [reply ykrt/src/jitstats.rs]: ah yeah, my bad.

**Pull request merged**